# **EDA of E-commerce Dataset with 30K Products:** 
### **Dated:** May 24nd, 2024
### **Dateset Link:** https://www.kaggle.com/datasets/mukuldeshantri/ecommerce-fashion-dataset/data
### Done by **Faizan Ahmad** (ma143faizan@gmail.com)

## **About Dataset:**
This dataset is a collection of 30000 women fashion products. 
Categories covered in this dataset is western wear, Indian wear, perfumes and fragrances, watches and nightwear. ✨

You can use this dataset to apply your data cleaning, visualization and analytical skills.
Column description is mentioned below:
- BrandName: Mentions the brand of the product
- Details: Deatils about the product
- Size: Sizes available
- MRP: This is max retail price
- SellPrice: This is the price after discount
- Category: Category of the product


In [186]:
# importing the libraries and dataset
import pandas as pd
import numpy as np
path = "FashionDataset.csv"
df = pd.read_csv(path)

In [187]:
# Checking the dataset 
df.head()

Unnamed: 0    BrandName                                            Deatils  \
0           0         life  solid cotton blend collar neck womens a-line d...   
1           1         only  polyester peter pan collar womens blouson dres...   
2           2      fratini  solid polyester blend wide neck womens regular...   
3           3  zink london  stripes polyester sweetheart neck womens dress...   
4           4         life  regular fit regular length denim womens jeans ...   

                                     Sizes       MRP SellPrice Discount  \
0  Size:Large,Medium,Small,X-Large,X-Small  Rs\n1699       849  50% off   
1                         Size:34,36,38,40  Rs\n3499      2449  30% off   
2              Size:Large,X-Large,XX-Large  Rs\n1199       599  50% off   
3          Size:Large,Medium,Small,X-Large  Rs\n2299      1379  40% off   
4                   Size:26,28,30,32,34,36  Rs\n1699       849  50% off   

            Category  
0  Westernwear-Women  
1  Westernwear-Women  
2  Westernwear-Women  
3  Westernwear-Women  
4  Westernwear-Women

In [188]:
df.tail()

Unnamed: 0  BrandName  \
30753          21  swarovski   
30754          22        Nan   
30755          23     jewelz   
30756          24    estelle   
30757          25    estelle   

                                                 Deatils Sizes       MRP  \
30753            crystal stylish womens rodhium earrings   Nan       Nan   
30754                                                Nan   Nan       Nan   
30755                 ethnic gold plated jhumki earrings   Nan  Rs\n1839   
30756  womens gold plated double line fancy white and...   Nan       Nan   
30757  womens gold plated bridge designer mangalsutra...   Nan       Nan   

      SellPrice Discount         Category  
30753      8950      Nan  Jewellery-Women  
30754       Nan      Nan  Jewellery-Women  
30755       643  65% off  Jewellery-Women  
30756      2799      Nan  Jewellery-Women  
30757      1899      Nan  Jewellery-Women

In [189]:
# All the unique values in each column
df_columns = df.columns
for column in df_columns:
    print(f"{column}'s Unique Values: {df[column].unique()}")
    print("--------------------------------------------------")

Unnamed: 0's Unique Values: [ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
 24 25]
--------------------------------------------------
BrandName's Unique Values: ['life' 'only' 'fratini' 'zink london' 'kraus' 'rare' 'van heusen' 'Nan'
 'stop' 'zink z' 'cover story' 'infuse' 'allen solly' 'altlife' 'madame'
 'iti' 'levis' 'and' 'faballey' 'latin quarters' 'sera' 'adidas' 'pepe'
 'indya' 'juniper' 'lovegen' 'vero moda' 'forever new' 'magre' 'insense'
 'calvin klein underwear' 'reebok' 'marie claire' 'enamor' 'spykar'
 'get wrapped' 'campus sutra' 'dolce crudo' 'emblaze' 'global desi'
 'jealous 21' 'jockey' 'crimsoune club' 'haute curry' 'proline' 'puma'
 'ivy' 'vh flex' 'myshka' 'femina flaunt' 'w' '109f' 'gipsy' 'samshek'
 'is.u' 'miss chase' 'ira soleil' 'fila' 'kami kubi' 'travel blue'
 'varanga' 'de moza' 'u.s. polo assn.' 'vip' 'janasya' 'skechers'
 'lee cooper' 'aarke' 'kashish' 'pepe kids' 'kenneth cole' 'zingg'
 'ayesha' 'elliza donatein' 'erotissch' 'bib

In [190]:
# info about the data to check for shape, column names, column dtypes, and missing values   
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30758 entries, 0 to 30757
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Unnamed: 0  30758 non-null  int64 
 1   BrandName   30758 non-null  object
 2   Deatils     30758 non-null  object
 3   Sizes       30758 non-null  object
 4   MRP         30758 non-null  object
 5   SellPrice   30758 non-null  object
 6   Discount    30758 non-null  object
 7   Category    30758 non-null  object
dtypes: int64(1), object(7)
memory usage: 1.9+ MB


## **Here's what I was able to understand (Things to do):**
- **It seems like there are NAN values but these are not being considered as Null values rather treated as Strings. Convert these NAN values into np.nan** ✔
- **The column "Unnamed: 0" is like index column in the data, but it has some issue because it is repeating numbers from 0 to 25; we can drop it.** ✔
- **The "Deatils" column should be renamed to "Details".** ✔
- **In the "Sizes" column, we can remove initial "Size:" word in each row if it exists.** ✔
- **From "MRP" column, Rs\n should be dropped from each row.** ✔
- **We can remove % off from "Discount" column.** ✔
- **If we take a look at the dataset's description on Kaggle, it clearly says "Dataset containing over 30K products of women's fashion", so we can remove -Women from the "Category" column.** ✔
- **We need to convert "Sell Price", "MRP" and "Discount" columns into numbers instead of strings.** ✔

In [191]:
# Dropping the Unnamed: 0 column
df.drop(columns="Unnamed: 0", inplace=True)

In [192]:
# Renaming the Deatils column to Details column
df.rename({"Deatils":"Details"}, inplace=True, axis=1)

In [193]:
# Removing unnecessary text from the columns
df.Sizes.replace({"Size:":""}, inplace=True, regex=True)
df.MRP.replace({"Rs\n":""}, inplace=True, regex=True)
df.Discount.replace({r"% off":""}, inplace=True, regex=True)
df.Category.replace({r"-Women":""}, inplace=True, regex=True)

In [194]:
# Converting the Nan strings into Np.NaN from the dataset, so they can be treated as null values
df.replace({"Nan":np.nan}, inplace=True)

In [195]:
# Converting the MRP, SellPrice, and Discount columns to numeric
df[["MRP","SellPrice","Discount"]] = df[["MRP","SellPrice","Discount"]].apply(pd.to_numeric)

In [196]:
# Checking random 25 samples from the dataset to see what else is needed to be done
df.sample(25)

BrandName                                            Details  \
10151               adidas  solid polyester round neck womens regular fit ...   
14971              varanga  printed cotton blend round neck womens ethnic ...   
13236              juniper  womens indigo cambric floral placement printed...   
21133              insense  printed cotton regular fit womens top and shor...   
21734               jockey  non-wired fixed straps non padded womens every...   
28723                kenzo      womens flower by eau de parfum 100 ml - multi   
20387           vastranand  structured zari cotton blend festive wear wome...   
17382           ira soleil  womens pink poly cotton cap sleeve printed kur...   
7835                  puma  polyester mid rise regular fit womens athleisu...   
28516              versace              womens yellow diamond deodorant 50 ml   
30208                  NaN                                                NaN   
20333           vastranand  solid zari silk blend festive wear womens regu...   
15960                  NaN                                                NaN   
12474               myshka  printed rayon round neck womens regular kurta ...   
5055           zink london     solid cotton sweetheart neck womens top - pink   
24488                  NaN                                                NaN   
17404                indya  geometric cotton regular fit womens maxi dress...   
4769                   and  solid round neck polyester women flared gown -...   
18013                indya  printed georgette womens full length sharara -...   
28121                casio     womens sheen round dial analogue watch - sx224   
15372          get wrapped  embroidered polyester blend regular length wom...   
20701           vastranand  solid zari linen festive wear womens regular s...   
19531           vastranand  floral zari cotton blend festive wear womens r...   
20030  varkala silk sarees  kanchipuram weave zari border and rich zari pa...   
27236                titan  womens raga mother of pearl dial metallic anal...   

                                                   Sizes     MRP  SellPrice  \
10151                                              Large  2499.0     1249.0   
14971                         Large,Medium,Small,X-Large     NaN     2099.0   
13236                 Large,Medium,Small,X-Large,X-Small  1799.0      989.0   
21133        Large,Medium,Small,X-Large,X-Small,XX-Large   899.0      449.0   
21734  32-B,32-C,34-B,34-C,34-D,36-B,36-C,36-D,38-B,3...     NaN      699.0   
28723                                                NaN     NaN     6250.0   
20387                                                NaN  5700.0     2280.0   
17382                                              Small     NaN     2499.0   
7835                  Large,Medium,Small,X-Large,X-Small     NaN     3999.0   
28516                                                NaN     NaN     2650.0   
30208                                                NaN     NaN        NaN   
20333                                                NaN  6500.0     2925.0   
15960                                                NaN     NaN        NaN   
12474                Large,Medium,Small,X-Large,XX-Large  1499.0      449.0   
5055                          Large,Medium,Small,X-Large  1399.0      699.0   
24488                                                NaN     NaN        NaN   
17404                                              Small     NaN     1850.0   
4769                                       8,10,12,14,16  4999.0     3999.0   
18013                Large,Medium,Small,X-Large,XX-Large     NaN     2600.0   
28121                                                NaN  9995.0     6996.0   
15372                                                NaN     NaN      349.0   
20701                                                NaN  3475.0     1911.0   
19531                                                NaN  4747.0     2136.0   
20030         

## **Things to do:**
- **There are NaN values in the dataset. Drop all such rows that have NaN in columns in such a way that those rows are useless to us.**
- **Imputing values in columns based on relevance. For example, we can calcualte MRP from Sale Price and Discount to impute it.**
- **Check for duplicate data.**

In [197]:
df.isnull().sum()

BrandName    1183
Details      1183
Sizes        7501
MRP          8208
SellPrice    1183
Discount     8208
Category        0
dtype: int64

- **It seems like there are 1183 rows in the dataset that don't have BrandName, Details and SellPrice. If the NaN values are all in the same rows then those rows become completely useless to us and we can drop them.** 
- **For some reason MRP and Discount has the most amount of null values, and maybe one of the reasons for this can the that it has nan value where the discount is 0%, so we can replace MRP with Sell Price for those particular rows.** 

In [198]:
# new df excluding the Category column
df_1 = df.drop(columns="Category")
df_1[df_1.isnull().all(axis=1)]

BrandName Details Sizes  MRP  SellPrice  Discount
22          NaN     NaN   NaN  NaN        NaN       NaN
48          NaN     NaN   NaN  NaN        NaN       NaN
74          NaN     NaN   NaN  NaN        NaN       NaN
100         NaN     NaN   NaN  NaN        NaN       NaN
126         NaN     NaN   NaN  NaN        NaN       NaN
...         ...     ...   ...  ...        ...       ...
30650       NaN     NaN   NaN  NaN        NaN       NaN
30676       NaN     NaN   NaN  NaN        NaN       NaN
30702       NaN     NaN   NaN  NaN        NaN       NaN
30728       NaN     NaN   NaN  NaN        NaN       NaN
30754       NaN     NaN   NaN  NaN        NaN       NaN

[1183 rows x 6 columns]

In [199]:
# Dropping all such rows from the df where all the values are missing
df.dropna(subset=["BrandName", "Details", "Sizes", "MRP", "SellPrice", "Discount"], how="all", inplace=True)

In [200]:
df.isnull().sum()

BrandName       0
Details         0
Sizes        6318
MRP          7025
SellPrice       0
Discount     7025
Category        0
dtype: int64

In [201]:
# Seeing rows that have MRP and Discount as missing values
df[df.MRP.isnull() & df.Discount.isnull()]

BrandName                                            Details  \
28                 zink z      solid cotton blend v neck womens top - maroon   
36                fratini  printed polyester blend round neck womens a-li...   
61                  levis   graphic cotton round neck womens t-shirt - brown   
69                  levis  tapered fit regular length cotton womens jeans...   
92                fratini  sanya malhotra regular fit regular length cott...   
...                   ...                                                ...   
30751           swarovski   constella stylish crystal white pierced earrings   
30752  shaya by caratlane               the wing woman clover charm necklace   
30753           swarovski            crystal stylish womens rodhium earrings   
30756             estelle  womens gold plated double line fancy white and...   
30757             estelle  womens gold plated bridge designer mangalsutra...   

                                     Sizes  MRP  SellPrice  Discount  \
28      Large,Medium,Small,X-Large,X-Small  NaN     1299.0       NaN   
36     Large,Medium,Small,X-Large,XX-Large  NaN     1799.0       NaN   
61              Large,Medium,Small,X-Small  NaN     1199.0       NaN   
69                          26,28,30,32,34  NaN     3799.0       NaN   
92                       26,28,30,32,34,36  NaN     1299.0       NaN   
...                                    ...  ...        ...       ...   
30751                                  NaN  NaN     8950.0       NaN   
30752                                  NaN  NaN     1250.0       NaN   
30753                                  NaN  NaN     8950.0       NaN   
30756                                  NaN  NaN     2799.0       NaN   
30757                                  NaN  NaN     1899.0       NaN   

          Category  
28     Westernwear  
36     Westernwear  
61     Westernwear  
69     Westernwear  
92     Westernwear  
...            ...  
30751    Jewellery  
30752    Jewellery  
30753    Jewellery  
30756    Jewellery  
30757    Jewellery  

[7025 rows x 7 columns]

In [202]:
df.Discount.unique()

array([50., 30., 40., 10., 60., nan, 20., 41., 55., 73., 69., 47., 57.,
       70., 67., 33., 44., 35., 45., 65., 68., 17., 78., 25., 80., 64.,
       63., 32., 58., 26., 71., 75., 79., 61., 24., 62., 38., 43., 54.,
       23., 27., 42., 76., 72., 53., 12., 74., 56., 22., 66., 39., 15.,
       34., 85.,  5., 77.,  4., 51., 37., 48.,  0., 52., 29., 28., 36.,
       31.])

In [203]:
# rows in which discount = 0
df[df.Discount == 0]

BrandName                                  Details              Sizes  \
24994  shoetopia   synthetic womens casual sandals - gold  35,36,37,38,39,40   
25370  shoetopia  synthetic womens casual sandals - black  35,36,37,38,39,40   

         MRP  SellPrice  Discount  Category  
24994  999.0      995.0       0.0  Footwear  
25370  999.0      995.0       0.0  Footwear

**There are only two rows with Discount = 0% and we can see there isn't much of a difference between MRP and SellPrice.** 

**So, we can replace Discount NaN values to 0 and then impute MRP = SellPrice for all such rows.**

In [204]:
df.Discount.replace({np.nan:0.0}, inplace=True)
df.loc[df['Discount'] == 0, 'MRP'] = df['SellPrice']

In [205]:
df.isnull().sum()

BrandName       0
Details         0
Sizes        6318
MRP             0
SellPrice       0
Discount        0
Category        0
dtype: int64

**Now, all other missing values have been dealt with besides Sizes. We can have a look at the data to understand why are there missing values in Sizes and how can we deal with it.**

In [206]:
# data rows sample in which Sizes is missing
df[df.Sizes.isnull()].sample(25)

BrandName                                            Details  \
26969            fastrack     womens white dial analogue watch - g9827pp01cj   
14102             janasya  printed chiffon festive wear designer womens s...   
20157          vastranand  printed zari linen festive wear womens regular...   
20336          vastranand  structured zari silk blend festive wear womens...   
19081          vastranand  solid zari silk blend festive wear womens regu...   
29408        pretty women     alloy metal gold color western womens earrings   
28277              skagen  womens grenen lille ocean white dial plastic a...   
26983               casio  womens 40-49 mm sheen rose gold dial metal ana...   
5473                 only            polyester regular fit womens cap - blue   
20262          vastranand  solid zari cotton blend festive wear womens re...   
30557              ayesha          neon orange hair clip for girls and women   
19519          vastranand  solid zari linen festive wear womens regular s...   
26790  fastrack wearables  unisex reflex curv blue strap silicone digital...   
28228               casio  womens 38 mm g-shock blue dial resin digital w...   
28546             arcelia                womens crazy crush deodorant 200 ml   
26705               titan  womens 53.51 x 7.40 x 29.33 mm rose gold dial ...   
27464               timex  womens black dial metallic analogue watch - tw...   
29329              ayesha  womens asymmetrical pearl lock statement neckl...   
16640          vastranand  structured zari silk blend festive wear womens...   
29918                 e2o                        womens gold tassel earrings   
30044           belleziya  gold finish and pearl embellished floral necklace   
30116        pretty women                womens alloy crystal hoops - yellow   
20377          vastranand  solid zari cotton blend festive wear womens re...   
29624               zingg               women stylish white casual hairbands   
20186          vastranand  stripes zari linen festive wear womens regular...   

      Sizes      MRP  SellPrice  Discount     Category  
26969   NaN   2450.0     2450.0       0.0      Watches  
14102   NaN   2249.0      899.0      60.0   Indianwear  
20157   NaN   5500.0     2475.0      55.0   Indianwear  
20336   NaN   4775.0     2387.0      50.0   Indianwear  
19081   NaN   5500.0     2200.0      60.0   Indianwear  
29408   NaN    469.0      469.0       0.0    Jewellery  
28277   NaN  13495.0    13495.0       0.0      Watches  
26983   NaN  10495.0     7346.0      30.0      Watches  
5473    NaN   1299.0      909.0      30.0  Westernwear  
20262   NaN   4545.0     2272.0      50.0   Indianwear  
30557   NaN    198.0       99.0      50.0    Jewellery  
19519   NaN   3475.0     1737.0      50.0   Indianwear  
26790   NaN   6495.0     4495.0      31.0      Watches  
28228   NaN  12995.0    11695.0      10.0      Watches  
28546   NaN    295.0      295.0       0.0    Fragrance  
26705   NaN  10495.0     5247.0      50.0      Watches  
27464   NaN   3195.0     2556.0      20.0      Watches  
29329   NaN    798.0      399.0      50.0    Jewellery  
16640   NaN   4775.0     2387.0      50.0   Indianwear  
29918   NaN   1099.0      659.0      40.0    Jewellery  
30044   NaN    799.0      799.0       0.0    Jewellery  
30116   NaN    299.0      254.0      15.0    Jewellery  
20377   NaN   4545.0     2272.0      50.0   Indianwear  
29624   NaN    299.0      299.0       0.0    Jewellery  
20186   NaN   3475.0     1911.0      45.0   Indianwear

**The problem here is that we have missing values for all such products that can't have any size like bracelets, face masks etc.**

**In my opinion, it would be better to just leave such NaN values empty ...**

In [207]:
# checking for duplicate data
df[df.duplicated()]

BrandName                                            Details  \
24             life  regular fit regular length denim womens jeans ...   
148            life  printed cotton lycra round neck womens t-shirt...   
280            life  printed cotton lycra round neck womens t-shirt...   
356            life  skinny fit regular length cotton lycra womens ...   
533            life  embroidered cotton blend round neck womens top...   
...             ...                                                ...   
30689  pretty women     alloy metal gold color western womens earrings   
30716        ayesha         womens stone studded drop earrings - multi   
30731     swarovski              crystal stylish womens white earrings   
30735        jewelz                cotton multicolour girls hair bands   
30751     swarovski   constella stylish crystal white pierced earrings   

                                    Sizes     MRP  SellPrice  Discount  \
24                      26,28,30,32,34,36  1999.0      999.0      50.0   
148    Large,Medium,Small,X-Large,X-Small   699.0      349.0      50.0   
280    Large,Medium,Small,X-Large,X-Small   699.0      349.0      50.0   
356                     26,28,30,32,34,36   999.0      499.0      50.0   
533    Large,Medium,Small,X-Large,X-Small  1299.0      649.0      50.0   
...                                   ...     ...        ...       ...   
30689                                 NaN   499.0      499.0       0.0   
30716                                 NaN   698.0      349.0      50.0   
30731                                 NaN  8950.0     8950.0       0.0   
30735                                 NaN   999.0      349.0      65.0   
30751                                 NaN  8950.0     8950.0       0.0   

          Category  
24     Westernwear  
148    Westernwear  
280    Westernwear  
356    Westernwear  
533    Westernwear  
...            ...  
30689    Jewellery  
30716    Jewellery  
30731    Jewellery  
30735    Jewellery  
30751    Jewellery  

[1651 rows x 7 columns]

**Although in an E-commerce store, often a product is present on a website multiple times but that is the case if there are different sellers. Other case is when a same product is being sold at differnt prices, OR it is placed in different category OR it belongs to a different Brand OR based on different availability of Sizes...**

**I'll drop only those duplicate rows that have same data in all the rows ...**

In [208]:
df.drop_duplicates(inplace=True, keep="first")

In [209]:
df.shape

(27924, 7)

### Initially, when we got the dataset, it had **30758 rows** but after all the necessary cleaning, we are left with only **27924 rows**. 

## **EDA Goals:**
1. **To find the most expensive brands...** ✔
2. **To find the cheapest brands...** ✔
3. **Most expensive Products on the store...** ✔
4. **Cheapest Products on the store...** ✔
5. **Number of products in each Category...** ✔
6. **To find the Brand that offers highest discounts...**

In [217]:
# Most Expensive Brands ...
brands_sort = df.groupby("BrandName").MRP.max().sort_values(ascending=False)
print("Top 10 Most Expensive Brands are:",brands_sort.head(10))
print("--------------------------------------------------")
print("Top 10 Cheapest Brands are:",brands_sort.tail(10))
print("--------------------------------------------------")


Top 10 Most Expensive Brands are: BrandName
versace               142500.0
tissot                 48500.0
michael kors           40995.0
emporio armani         39995.0
titan                  31995.0
swarovski              31000.0
coach                  28495.0
alexandre christie     25995.0
fossil                 24995.0
just cavalli           24900.0
Name: MRP, dtype: float64
--------------------------------------------------
Top 10 Cheapest Brands are: BrandName
pretty women               599.0
plum                       525.0
vip                        499.0
travel blue                499.0
hidesign                   495.0
superdry                   399.0
zingg                      299.0
black panther              299.0
beverly hills polo club    250.0
ariana                     200.0
Name: MRP, dtype: float64
--------------------------------------------------


In [236]:
# Creating plotly express graph for the most expensive brands
import plotly.express as px
fig = px.bar(brands_sort.head(25), x=brands_sort.head(25).index, y="MRP", title="Top 25 Most Expensive Brands", color="MRP")
fig1 = px.bar(brands_sort.tail(25), x=brands_sort.tail(25).index, y="MRP", title="Top 25 Cheapest Brands", color="MRP")
fig.show()
fig1.show()

In [219]:
# Most Expensive and Cheapest Products ...
products_sort = df.groupby("Details").MRP.max().sort_values(ascending=False)
print("Top 10 Most Expensive Products are:",products_sort.head(10))
print("--------------------------------------------------")
print("Top 10 Cheapest Products are:",products_sort.tail(10))
print("--------------------------------------------------")

Top 10 Most Expensive Products are: Details
womens 36 mm greca icon silver white guilloch dial stainless steel analog watch - vez600621      142500.0
womens 34 mm meander white dial steel ion plating 5n analogue watch - velw00620                  137900.0
unisex 43 mm pink dial bracelet analogue watch - vez300621                                       108000.0
womens 36 mm versace essential black dial stainless steel analogue watch - vek400621              99100.0
womens 35 mm v-motif white-silver dial stainless steel analog watch - vere02420                   91900.0
womens v- motif gold-tone yellow gold ion plated analogue watch                                   91100.0
womens 36 mm blue dial bracelet analogue watch - vek400821                                        55700.0
mens 39.3 mm le locle powermatic 80 blue dial stainless steel analogue watch - t0064071104300     48500.0
womens white pop chic lady 36mm analogue watch - wvevd00519n                                      48500.0
wo

In [234]:
# Creating plotly express graph for the most expensive products
fig = px.bar(products_sort.head(15), x="MRP", y=products_sort.head(15).index, title="Top 15 Most Expensive Products", color = "MRP")
fig1 = px.bar(products_sort.tail(15), x="MRP", y=products_sort.tail(15).index, title="Top 15 Cheapest Products", color = "MRP")
fig.show()
fig1.show()

In [239]:
# Number of Products in each Category ...
category_count = df.Category.value_counts()
print("Number of Products in each Category are:",category_count)


Number of Products in each Category are: Category
Westernwear           9724
Indianwear            8906
Lingerie&Nightwear    3154
Footwear              2356
Watches               1696
Jewellery             1614
Fragrance              474
Name: count, dtype: int64


In [255]:
# Creating plotly express graph for the number of products in each category without color bar
fig = px.bar(category_count, x=category_count.index, y=category_count.values, title="Number of Products in each Category")
fig.show()

In [261]:
# Brands that offers the highest discounts rates ...
discounts = df.groupby("BrandName").Discount.max().sort_values(ascending=False)
print("Top 10 Brands that offer the Highest discounts are:",discounts.head(10))

Top 10 Brands that offer the Highest discounts are: BrandName
myshka                 85.0
109f                   80.0
lovegen                80.0
fusion beats           80.0
zink london            80.0
iti                    78.0
catwalk                78.0
nayomi                 77.0
varkala silk sarees    75.0
juniper                75.0
Name: Discount, dtype: float64


In [262]:
# Creating plotly express graph for the brands that offer the highest discount rates
fig = px.bar(discounts.head(15), x=discounts.head(15).index, y="Discount", title="Top 15 Brands that offer the Highest discounts", color="Discount")
fig.show()